guard/tests/validate.rs

```diff
@@ -328,44 +365,425 @@ mod validate_tests {
+        let mut expected = String::new();
+
+        file.read_to_string(&mut expected).unwrap();
+
+        assert_eq!(expected, result);
+        assert_eq!(StatusCode::SUCCESS, status_code);
+    }
+
+    #[test]
+    fn test_rules_with_data_from_stdin_fail() {
+        let file = File::open(
+            "resources/validate/data-dir/s3-public-read-prohibited-template-non-compliant.yaml",
+        )
+        .expect("failed to find mocked file ");
+        let mut reader = Reader::new(ReadFile(file));
+        let mut writer = Writer::new(WBVec(vec![]), WBVec(vec![]));
+        let status_code = ValidateTestRunner::default()
+            .rules(vec!["rules-dir/s3_bucket_public_read_prohibited.guard"])
+            .verbose(true)
+            .run(&mut writer, &mut reader);
+
+        let mut file =
+            File::open("resources/validate/output-dir/payload_verbose_non_compliant.out").unwrap();
+        let result = writer.stripped().unwrap();
+        let mut expected = String::new();
+
+        file.read_to_string(&mut expected).unwrap();
+
+        assert_eq!(result, expected);
+        assert_eq!(StatusCode::PARSING_ERROR, status_code);
+    }
+
+    #[test]
+    fn test_rules_with_data_from_stdin_verbose_previous_engine_json_fail() {
+        let file = File::open(
+            "resources/validate/data-dir/s3-public-read-prohibited-template-non-compliant.yaml",
+        )
+        .expect("failed to find mocked file ");
+        let mut reader = Reader::new(ReadFile(file));
+        let mut writer = Writer::new(WBVec(vec![]), WBVec(vec![]));
+        let status_code = ValidateTestRunner::default()
+            .rules(vec!["rules-dir/s3_bucket_public_read_prohibited.guard"])
+            .verbose(true)
+            .previous_engine(true)
+            .print_json(true)
+            .run(&mut writer, &mut reader);
+
+        let result = writer.stripped().unwrap();
+        let mut file =
+            File::open("resources/validate/output-dir/payload_verbose_json_non_compliant.out")
+                .unwrap();
+        let mut expect = String::new();
+        file.read_to_string(&mut expect).unwrap();
+
+        assert_eq!(expect, result);
+        assert_eq!(StatusCode::PARSING_ERROR, status_code);
+    }
+
+    #[test]
+    fn test_payload_verbose_yaml_compliant() {
+        let file = File::open(
+            "resources/validate/data-dir/s3-public-read-prohibited-template-compliant.yaml",
+        )
+        .expect("failed to find mocked file ");
+        let mut reader = Reader::new(ReadFile(file));
+        let mut writer = Writer::new(WBVec(vec![]), WBVec(vec![]));
+        let status_code = ValidateTestRunner::default()
+            .rules(vec!["rules-dir/s3_bucket_public_read_prohibited.guard"])
+            .verbose(true)
+            .output_format(Some("yaml"))
+            .run(&mut writer, &mut reader);
+        let result = writer.stripped().unwrap();
+
+        let mut file =
```

Comment:
Possible usage of `assert_output_from_file_eq` here (I have repeated this comment a couple times, but have not covered all occurrences of using a File::open() instead, if you end up addressing this comment be sure to replace them all)